pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -352,6 +352,24 @@ public TreeSet<String> getPhysicalColumnNames() {
     return physicalColumnNames;
   }
 
+  /**
+   * Returns a new schema containing only physical columns.
+   *
+   * All properties but the fields are the same.
+   * All field attributes are a shallow copy without the virtual column.
+   */
+  public Schema withoutVirtualColumns() {
```

Comment:
While there might not be any current use case for it, I think it'd be good to also copy over the primary key columns to the new schema?

Reply:
You are right. Added (and added some tests)